### DISCUSSION TITLE
Is there any documentation which helps to change the target version to 34 in mobile app?

### DISCUSSION THREAD

**Discussion by sayali2523** [Sunbird-Ed/Community#745] [Documentation]
In our mobile application, we recently updated the target SDK version to 34 as required by Google Play’s new guidelines. The update to SDK 34 introduced changes to permission handling, storage access, and other platform-specific behaviors that affected certain functionalities of our app, specifically the ECML player used within the Sunbird mobile SDK.
The ECML player relies on the unzipping of .ecar or .zip files that contain learning content (ECML files). On Android 14, the unzip process fails, preventing the content from being loaded. This worked correctly on Android 13 and below, but Android 14’s new file access rules have caused issues.
@swayangjit @rajeevsathish @nidhikachhawa24

**Comment by HarishGangula** [comment]
@sayali2523 Share the details of the issue which is causing along with the error logs will help in resolving the issue

**Comment by sayalimore16** [comment]
<img width="666" alt="Screenshot 2024-09-12 at 5 37 36 PM" src="https://github.com/user-attachments/assets/70b2f39b-52d7-4936-821d-0cac28798f82">
We are getting the above error so we tried download and read the content. it is working till android 13 but in android 14 nested zip folder is getting created which we are unable to unzip.
<img width="247" alt="Screenshot 2024-09-12 at 5 41 28 PM" src="https://github.com/user-attachments/assets/ca21d329-04f1-46af-bb75-c8995461225a">

**Reply by HarishGangula** [reply]
@sayali2523 It might be the issue with unzip plugin configurations check the plugin compatibility with android 14.

**Reply by sayali2523** [reply]
@HarishGangula we used Cordova zip plugin this plugin is work for h5p content which has same ecar file extension. only for ECML it fails in android 14.
we tried both below plugins. 
import { Zip } from '@ionic-native/zip/ngx';
import JSZip from 'jszip';

**Reply by HarishGangula** [reply]
@sayali2523  Mobile app uses https://github.com/Sunbird-Ed/jjdltc-cordova-plugin-zip for this, You enhance this plugin to support for the android 14 version

**Reply by sayali2523** [reply]
Hi @HarishGangula 
We initially used the same plugin but encountered the same error, which is why we tried the alternative plugin mentioned above. Could you please let us know if there have been any updates to this plugin, specifically for Android 14 support with ECML files? We are currently using the master branch.

**Reply by HarishGangula** [reply]
@sayali2523 We have not done any changes for the migration to Android 14, It will be helpful if you can provide the PR for the same.

**Reply by siddhishinde0723** [reply]
@sayali2523 Can you please help us with the PR, or can we connect for the same?

**Reply by samarthtrigyn** [reply]
> @sayali2523 We have not done any changes for the migration to Android 14, It will be helpful if you can provide the PR for the same.

Hi @HarishGangula what is the alternative that you are using incase of cordova to comply with the android 14, SDK 34 policies? Or the latest android app does not support android 14?

**Reply by HarishGangula** [reply]
@samarthtrigyn  We will design and plan to move to capacitor in up coming releases, PRs are welcome

**Comment by nidhikachhawa24** [comment]
@sayali2523 

Can you please help us with the PR, or can we connect for the same?
We need help here !!
